v105.0.0

yarneo opened the pull request:
In this major release we removed the deprecated themers: MDCTextFieldTypographyThemer, MDCOutlinedButtonThemer.

## Breaking changes

Deleted MDCTextFieldTypographyThemer and MDCOutlinedButtonThemer classes. Please use the theming extensions.
Learn more at https://github.com/material-components/material-components-ios/blob/develop/docs/theming.md#migration-guide-themers-to-theming-extensions

## Component changes

### ActionSheet

* [Migrates Action sheet to MDCAvailability. (#9733)](https://github.com/material-components/material-components-ios/commit/652077c779b06fc6b44e9bbae237f47e083c053c) (Bryan Oltman)

### ActivityIndicator

* [Migrates activity indicator to MDCAvailability. (#9735)](https://github.com/material-components/material-components-ios/commit/28a7fac63f7ee5f1787ab56190c8a579532810d6) (Bryan Oltman)

### BottomNavigation

* [Migrates Bottom Navigation to MDCAvailability. (#9717)](https://github.com/material-components/material-components-ios/commit/02efdd188990305ff298fbad25fa48634974b1b4) (Randall Li)

### Buttons

* [Fix tvOS target bug. (#9705)](https://github.com/material-components/material-components-ios/commit/95518f48d4fe6c10e13cbdcfcc3a1c863697298d) (Randall Li)
* [MDCButton. (#9667)](https://github.com/material-components/material-components-ios/commit/1d8d647f1f534de4366c8722992b67858d7ba444) (copybara-service[bot])
* [Delete deprecated MDCOutlinedButtonThemer (#9688)](https://github.com/material-components/material-components-ios/commit/1118f14124858fe9a258b0971393e33b3966d2bf) (Bryan Oltman)

### Cards

* [Migrates cards to MDCAvailability. (#9728)](https://github.com/material-components/material-components-ios/commit/b677a3629e180a89f1e6a15256b45790ca794af5) (Randall Li)

### Chips

* [Migrates Chips to MDCAvailability. (#9713)](https://github.com/material-components/material-components-ios/commit/33fbdfc3be62b4ff78951898b9abe6fc59041db4) (Randall Li)

### Dialogs

* [Add snapshot tests for additional dialog configurations. (#9646)](https://github.com/material-components/material-components-ios/commit/e2bff768df18fc24f91857447ddb8992071e778f) (copybara-service[bot])
* [Adding accessory view vertical inset to private header (#9692)](https://github.com/material-components/material-components-ios/commit/ecddca8d58d8e02fb350d31277693dbaff56f952) (Galia Kaufman)
* [Adding an adjustable insets feature flag. Flag default is NO. Clients must enable flag if they want to customize the inset. This flag has no effect yet in this CL. Adjustable layout logic will be introduced in followup CLs. (#9648)](https://github.com/material-components/material-components-ios/commit/0c8647cfca40fc04aa9da1ff7dde30efbe6456dc) (copybara-service[bot])
* [Delete Dialogs OWNERS file (#9697)](https://github.com/material-components/material-components-ios/commit/5884725f8746bfaecbc02ed509cb70eac42c5428) (Randall Li)
* [Fix custom title icon view layout - top inset (#9721)](https://github.com/material-components/material-components-ios/commit/3b89aecee05c95d49f331ce77427cc5731fbea99) (Galia Kaufman)
* [Migrate dialogs to availability (#9743)](https://github.com/material-components/material-components-ios/commit/88b5767bdb200438108c46f9eae96a8695ef91f7) (Andrew Overton)
* [Refactor action frame calculations to accommodate adjustable insets (5 seconds ago) (#9656)](https://github.com/material-components/material-components-ios/commit/a40a4eb171994b72de473b974131350bed870b1b) (copybara-service[bot])
* [Refactor content frame calculations to accommodate adjustable insets  (#9710)](https://github.com/material-components/material-components-ios/commit/05e38ef7387ff7bb455978031339ecda3ae19893) (Galia Kaufman)
* [Refactor title frame calculations to accommodate adjustable insets (#9709)](https://github.com/material-components/material-components-ios/commit/6011e001f6cfc28ea28591d098a14b9cff4ca78f) (Galia Kaufman)
* [Refactor title-icon, title-icon-view frames calculations to accommodate adjustable insets (#9657)](https://github.com/material-components/material-components-ios/commit/b3ba7e6d2c9deeb7c155f2242cad3c10c15f3e39) (copybara-service[bot])
* [Update adjustable insets comments (#9694)](https://github.com/material-components/material-components-ios/commit/60a531ca2ad9604c7559be0241aabec874cb87a4) (Galia Kaufman)
* [Update adjustable insets values (#9695)](https://github.com/material-components/material-components-ios/commit/735d2d28183cf977eb9508ac913a234f8e4d5bc3) (Galia Kaufman)

### FlexibleHeader

* [Added property for minimumHeaderViewHeight (#9649)](https://github.com/material-components/material-components-ios/commit/8e782d43a2d0a8033f0c65809c9dd09624ff3dbd) (copybara-service[bot])
* [Migrates Flexible Header to MDCAvailability. (#9744)](https://github.com/material-components/material-components-ios/commit/d962e50cf2564d07a6b823bf02170bdd7fb38f20) (Bryan Oltman)

### Ink

* [Migrates Ink to MDCAvailability. (#9741)](https://github.com/material-components/material-components-ios/commit/f168dd2806d5b9a069f08dd715cb8dc895db996e) (Bryan Oltman)

### NavigationDrawer

* [Delete NavigationDrawer OWNERS file (#9696)](https://github.com/material-components/material-components-ios/commit/43706fa2f9c919421867e442656b24c6dd762da1) (Randall Li)

### Slider

* [Migrates Slider to MDCAvailability. (#9704)](https://github.com/material-components/material-components-ios/commit/d03c34293fb3935e2d24f306e5add2f81741ea21) (Randall Li)

### Snackbar

* [Migrates Snackbar to MDCAvailability. (#9739)](https://github.com/material-components/material-components-ios/commit/1a221fe8b02aa2c1df77ad91a00f544c7eb1756a) (Bryan Oltman)

### Tabs

* [Migrates Tabs to MDCAvailability. (#9716)](https://github.com/material-components/material-components-ios/commit/549e52668e3525bc41e728ccf3e158574951bdd1) (Randall Li)

### TextFields

* [Delete deprecated MDCTextFieldTypographyThemer (#9689)](https://github.com/material-components/material-components-ios/commit/86b4fbcb2c2074b9e60dfb7987a3a1dc92c37937) (Bryan Oltman)

### private/Color

* [Make private/color use availability  (#9738)](https://github.com/material-components/material-components-ios/commit/1cc8bb781d6142452f40d8265664077ba7c85710) (Andrew Overton)

### private/TextControlsPrivate

* [Migrates Text Controls to MDCAvailability (#9720)](https://github.com/material-components/material-components-ios/commit/d89fa3fb11b97727265254afac91c2f9be6afb21) (Andrew Overton)
* [Move static const assignment to MDCTextControl.m (#9701)](https://github.com/material-components/material-components-ios/commit/2d8d6a082fbf046df5b108b03e4566fb1fe91ea8) (Andrew Overton)
* [Changes from text-area-feature-branch with additional podspec and BUILD file changes (#9711)](https://github.com/material-components/material-components-ios/commit/e6ef298bdff1f25005f013777f9e534cb4b6a6d4) (Andrew Overton)

### schemes/Color

* [Migrates Schemes to MDCAvailability (#9736)](https://github.com/material-components/material-components-ios/commit/ae6d6d9a908c13f0466ad566ba7763925739dd49) (Andrew Overton)

bryanoltman committed:
[Buttons] Delete deprecated MDCOutlinedButtonThemer (#9688)

MDCOutlinedButtonThemer is deprecated and has no internal usage.

Fixes #9687

bryanoltman committed:
[TextFields] Delete deprecated MDCTextFieldTypographyThemer (#9689)

MDCTextFieldTypographyThemer is deprecated and has no internal usage.

Fixes #9661

copybara-service[bot] committed:
[Dialogs] Adding an adjustable insets feature flag. Flag default is NO. Clients must enable flag if they want to customize the inset. This flag has no effect yet in this CL. Adjustable layout logic will be introduced in followup CLs. (#9648)

Adding an adjustable insets feature flag to the alert view private header. Default is NO.(b/132948073)

bryanoltman committed:
[TextFields] Update snapshot goldens (#9691)

New golden images were generated using `./.kokoro -d bazel -t //components/TextFields/...` and Xcode 10.3. They're replacing faulty images generated by recording in Xcode.

Partial rollback of https://github.com/material-components/material-components-ios/pull/9666

yarneo committed:
fix snapshot tests for CI (#9698)

New golden images were generated using ./.kokoro -d bazel -t //components/FeatureHighlight/...  and ./.kokoro -d bazel -t //components/Snackbar/... using Xcode 10.3. They're replacing faulty images generated by recording in Xcode.

copybara-service[bot] committed:
[FlexibleHeader] Added property for minimumHeaderViewHeight (#9649)

Create new property, minimumHeaderViewHeight, to describe an amount of header to remain on the screen after shiftBehavior is enabled and header has been shifted/scrolled off screen.

randallli committed:
merged release-candidate

randallli committed:
Merge branch 'stable' into release-candidate

copybara-service[bot] committed:
[Component] MDCButton. (#9667)

[Component] MDCButton.

Corrected a spelling mistake in the error response for touch target dimensions.

galiak11 committed:
[Dialogs] Adding accessory view vertical inset to private header (#9692)

Adding accessory view vertical inset to private header, with a default value of 20.0, which is a backward compatible value (b/132948073).

randallli committed:
[buttons] Fix tvOS target bug. (#9705)

PiperOrigin-RevId: 294713878

randallli committed:
Delete NavigationDrawer OWNERS file (#9696)

This file has the PII of Googlers.

https://github.com/material-components/material-components-ios/blob/stable/CODEOWNERS is github friendly version of the same information.

randallli committed:
Delete Dialogs OWNERS file (#9697)

This file has the PII of Googlers.

https://github.com/material-components/material-components-ios/blob/stable/CODEOWNERS is github friendly version of the same information.

andrewoverton committed:
[TextControls] Changes from text-area-feature-branch with additional podspec and BUILD file changes (#9711)

Because CI wasn't working earlier this week I merged the initial text area PRs (https://github.com/material-components/material-components-ios/pull/9682 and https://github.com/material-components/material-components-ios/pull/9684) into a [feature branch](https://github.com/material-components/material-components-ios/tree/text-area-feature-branch) in the main repo. This PR contains those changes, as well as a commit to update the BUILD file and podspec (3da2803), a commit that makes it compile (6a06594), and a commit to run clang (078cd70). The changes in these last three commits have not been reviewed, unlike the ones from the PRs into the feature branch.

Related to #9407.

andrewoverton committed:
Remove test spec (#9718)

https://github.com/material-components/material-components-ios/pull/9711 causes the library to fail pod lib lint validations because there are currently no source files in the test spec. This will of course change very soon but in case there's a release before it does we should land this PR.

Related to #9407.

andrewoverton committed:
[Material][Availability] Migrates Text Controls to MDCAvailability (#9720)

This PR addresses the failures in https://github.com/material-components/material-components-ios/pull/9719, which was copybara'd from [cl/293702448](http://cl/293702448)

randallli committed:
[Material][Availability] Migrates Slider to MDCAvailability. (#9704)

[Material][Availability] Migrates Slider to MDCAvailability.

randallli committed:
Merge branch 'stable' into release-candidate

randallli committed:
merged release-candidate

andrewoverton committed:
Move static const assignment to MDCTextControl.m (#9701)

This change makes the static const in MDCTextControl.h a FOUNDATION_EXTERN const. 

Related to [b/143312111](http://b/143312111).

bryanoltman committed:
[Material][Availability] Migrates Action sheet to MDCAvailability. (#9733)

[Material][Availability] Migrates Action sheet to MDCAvailability.

bryanoltman committed:
[Material][Availability] Migrates activity indicator to MDCAvailability. (#9735)

* [Material][Availability] Migrates activity indicator to MDCAvailability.

PiperOrigin-RevId: 293702405

* update build file

Co-authored-by: Eric Lee <eric.lee.edl@gmail.com>

andrewoverton committed:
[private] Make private/color use availability  (#9738)

Original CL: cl/293702453

bryanoltman committed:
[Material][Availability] Migrates Snackbar to MDCAvailability. (#9739)

[Material][Availability] Migrates Snackbar to MDCAvailability.

bryanoltman committed:
[Material][Availability] Migrates Ink to MDCAvailability. (#9741)

[Material][Availability] Migrates Ink to MDCAvailability.

andrewoverton committed:
[Material][Availability] Migrates Schemes to MDCAvailability (#9736)

Original CL: cl/293702425

bryanoltman committed:
[Material][Availability] Migrates Flexible Header to MDCAvailability. (#9744)

[Material][Availability] Migrates Flexible Header to MDCAvailability.

randallli committed:
Add @jakerockland to CODEOWNERS: NavigationDrawer (#9715)

andrewoverton committed:
[Dialogs] Migrate dialogs to availability (#9743)

Original CL: cl/293702412

randallli committed:
[Material][Availability] Migrates Bottom Navigation to MDCAvailability. (#9717)

[Material][Availability] Migrates Bottom Navigation to MDCAvailability.

randallli committed:
[Material][Availability] Migrates Chips to MDCAvailability. (#9713)

[Material][Availability] Migrates Chips to MDCAvailability.

randallli committed:
[Material][Availability] Migrates Tabs to MDCAvailability. (#9716)

[Material][Availability] Migrates Tabs to MDCAvailability.

randallli committed:
[Material][Availability] Migrates Action sheet to MDCAvailability. (#9732)

[Material][Availability] Migrates Action sheet to MDCAvailability.

randallli committed:
[Material][Availability] Migrates cards to MDCAvailability. (#9728)

[Material][Availability] Migrates cards to MDCAvailability.

bryanoltman committed:
Fix invalid podspec (#9745)

galiak11 committed:
[Dialogs] Fix custom title icon view layout - top inset (#9721)

Fix for the top layout inset of a custom title icon view, to allow for full-bleed images.

andrewoverton committed:
Remove unneeded subspec (#9748)

This change removes an empty subspec that was causing pod lib lint to fail.
Closes #9747.

galiak11 committed:
[Dialogs] Update adjustable insets comments (#9694)

Minor fixes to description of the private properties (b/148802180).

galiak11 committed:
[Dialogs] Update adjustable insets values (#9695)

Update adjustable insets values, so they can be used for adjustable insets that match current fixed insets (b/148802180).

galiak11 committed:
[Dialogs] Refactor title frame calculations to accommodate adjustable insets (#9709)

Compute adjustable insets for the title frame. Both adjustable and fixed insets mode are supported. Adjustable insets are hidden behind a feature flag (b/148802180).

galiak11 committed:
[Dialogs] Refactor content frame calculations to accommodate adjustable insets  (#9710)

Refactor content frame calculations to enable both fixed and adjustable insets layout. Use enableAdjustableInsets to toggle between adjustable and fixed insets (b/148802180).

copybara-service[bot] committed:
[Dialogs] Refactor action frame calculations to accommodate adjustable insets (5 seconds ago) (#9656)

Co-authored-by: Galia Kaufman <galiak@google.com>, Iryna Berezan <iberezan@google.com>

Enable adjustable insets for Dialogs actions. Both fixed and adjustable layout are supported. The adjustable layout is hidden behind a feature flag (b/148802180).

copybara-service[bot] committed:
[Dialogs] Refactor title-icon, title-icon-view frames calculations to accommodate adjustable insets (#9657)

Co-authored-by: Galia Kaufman <galiak@google.com>, Iryna Berezan <iberezan@google.com>

Use titleIconInsets to customize the insets around the title-icon or title-icon-view (b/148802180),

copybara-service[bot] committed:
[Dialogs] Add snapshot tests for additional dialog configurations. (#9646)

Co-authored-by: Galia Kaufman <galiak@google.com>

[Dialogs] Add snapshot tests for various configurations of elements within Dialogs (b/148802180).

yarneo committed:
Automatic changelog preparation for release.

yarneo committed:
Hand-modified CHANGELOG.md API diff.

yarneo committed:
Bumped version number to 105.0.0.

galiak11 commented on CHANGELOG.md:
```diff
@@ -1,3 +1,104 @@
+
+### Dialogs
+
+* [Add snapshot tests for additional dialog configurations. (#9646)](https://github.com/material-components/material-components-ios/commit/e2bff768df18fc24f91857447ddb8992071e778f) (copybara-service[bot])
```
Why some PRs are not attributed to the author(s)?

yarneo replied:
they were copied via the copybara bot from internal CLs which is why they show like that.

yarneo replied:
Changed to your name in these PRs

galiak11 commented on CHANGELOG.md:
```diff
@@ -1,3 +1,104 @@
+* [Adding accessory view vertical inset to private header (#9692)](https://github.com/material-components/material-components-ios/commit/ecddca8d58d8e02fb350d31277693dbaff56f952) (Galia Kaufman)
+* [Adding an adjustable insets feature flag. Flag default is NO. Clients must enable flag if they want to customize the inset. This flag has no effect yet in this CL. Adjustable layout logic will be introduced in followup CLs. (#9648)](https://github.com/material-components/material-components-ios/commit/0c8647cfca40fc04aa9da1ff7dde30efbe6456dc) (copybara-service[bot])
+* [Delete Dialogs OWNERS file (#9697)](https://github.com/material-components/material-components-ios/commit/5884725f8746bfaecbc02ed509cb70eac42c5428) (Randall Li)
+* [Fix custom title icon view layout - top inset (#9721)](https://github.com/material-components/material-components-ios/commit/3b89aecee05c95d49f331ce77427cc5731fbea99) (Galia Kaufman)
```
Can you attribute this to (Iryna Berezan) instead?

yarneo replied:
Done

yarneo committed:
update changelog

andrewoverton commented:
@yarneo I wouldn't mention the text areas, they are not ready for use yet.

bryanoltman committed:
[Dialogs] Add reference images for MDCAlertControllerConfigurationsTests (#9757)

Adds the missing reference images required for the tests introduced by 31c2f23 to pass.

yarneo committed:
updated changelog

yarneo committed:
snapshots
